File: js/src/trees/levelOrder.js
const levelOrder = (root) => {
	if (!root) return [];
	const res = [];
	const queue = [root];
	while (queue.length) {
		const level = [];
		const queueSize = queue.length;
		for (let i = 0; i < queueSize; i++) {
			const current = queue.shift();
			level.push(current.val);
			if (current.left) queue.push(current.left);
			if (current.right) queue.push(current.right);
		}
		res.push(level);
	}
	return res;
};

File: python/trees/levelOrder.py
from collections import deque

class Solution:
    def levelOrder(root):
        if not root:
            return []
        res = []
        queue = deque()
        queue.append(root)

        while queue:
            level = []
            n = len(queue)
            for i in range(n):
                current = queue.popleft()
                level.append(current.val)

                if current.left:
                    queue.append(current.left)
                if current.right:
                    queue.append(current.right)

            if level:
                res.append(level)
        return res
